_posts/pl/2022-02-16-porownanie-jezykow-cloud-functions.md

```diff
@@ -0,0 +1,146 @@
+
+Do automatyzacji wykorzystany został Terraform, za pomocą którego przygotowywane jest całe środowisko testowe. Wszystkie osadzane funkcje są definiowane w konfiguracji, dlatego w prosty sposób można uruchomić środowisko testujące wybrane języki oraz regiony.
+
+Testy czasów odpowiedzi zostały napisane w Gatlingu, który listę funkcji odczytuje z tej samej konfiguracji, przez co nie wymaga żadnej dodatkowej ingerencji. Testy zimnych startów wykonywane są natomiast bezpośrednio przez kod napisany w Scali, a wyniki wyświetlane są w formie tabeli ASCII.
+
+Kody wszystkich funkcji znajdują się w folderze `/functions` i są to podstawowe funkcje odpowiadające *"Hello World"*, takie same jak przykładowe funkcje tworzone z poziomu Cloud Console.
+
+Projekt znajduję się na GitHubie - [link do repozytorium.](https://github.com/Michuu93/google-cloud-function-comparison)
+
+## Metodyka testowania
+
+W testach wykorzystałem funkcje uruchomione w następujących środowiskach uruchomieniowych:
+
+- .NET Core 3.1
+- Go 1.13
+- Java 11
+- Node.js 14
+- PHP 7.4
+- Python 3.9
+- Ruby 2.7
+
+Każda funkcja posiadała maksymalnie jedną instancję, przydzieloną pamięć 128 MB i została uruchomiona w regionach:
+
+- `europe-west3` (Frankfurt, Germany, Europe)
+- `us-central1` (Council Bluffs, Iowa, North America)
+- `asia-east2` (Hong Kong, APAC)
+
+W celu porównania czasów odpowiedzi, każda funkcja była wywoływana przez 10 minut i 20 równoległych użytkowników. Ilość wykonanych żądań jest zatem uzależniona od samej funkcji oraz Gatlinga (a w zasadzie, to mojego laptopa).
+
+Testy zimnych startów zostały wykonane z zapewnieniem braku istnienia aktywnej instancji. Test polegał na wykonaniu 10 żądań do każdej z funkcji, a następnie porównaniu czasu pierwszej odpowiedzi do średniej arytmetycznej czasów pozostałych 9 odpowiedzi.
+
+Każdy test uruchomiłem dwa razy, o tej samej godzinie czasu polskiego dla wszystkich regionów oraz o tej samej godzinie czasu lokalnego w danym regionie. Wszystkie żądania były wykonywane z mojej stacji roboczej w Poznaniu.
+
+Przyjąłem nazewnictwo *język interpretowany* dla języków skryptowych oraz kompilowanych (nie korzystających z maszyny wirtualnej) oraz *język uruchamiany w maszynie wirtualnej* dla języków kompilowanych i uruchamianych w maszynie wirtualnej.
+
+# Czasy odpowiedzi
+
+<link href="{{ base.url | prepend: site.url }}/assets/css/tabs.css" rel="stylesheet" type="text/css">
+<link href="{{ base.url | prepend: site.url }}/assets/css/simple-datatables.css" rel="stylesheet" type="text/css">
+<script src="{{ base.url | prepend: site.url }}/assets/js/simple-datatables.js"></script>
+
+## Godziny uruchomienia testów
+
+- `Run #1` - środek tygodnia, 17-23 czasu polskiego
+- `Run #2` - środek tygodnia, 20:00 czasu lokalnego w danym regionie:
```

Comment:
może zamiast "środek tygodnia", po prostu środa? o ile to była środa :)

Reply:
no właśnie to była czasami środa, czasami wtorek/czwartek z tego co pamiętam :P